This is looking good so far. You have some trailing whitespaces that the linter doesn't like:

```
sympy/testing/tests/test_code_quality.py:236: in test_this_file
    assert False, message_space % (fname, idx + 1)
E   AssertionError: File contains trailing whitespace: /home/runner/work/sympy/sympy/sympy/physics/vector/frame.py, line 1403.
E   assert False
```

--- a/sympy/physics/vector/frame.py
+++ b/sympy/physics/vector/frame.py
@@ -163,7 +163,7 @@ class ReferenceFrame:
 
         >>> from sympy.physics.vector import ReferenceFrame
         >>> N = ReferenceFrame('N')
-        >>> N.unit
+        >>> N.u
         (N.x|N.x) + (N.y|N.y) + (N.z|N.z)
         >>> P = ReferenceFrame('P', indices=['1', '2', '3'])
         >>> P.u

--- a/sympy/physics/vector/frame.py
+++ b/sympy/physics/vector/frame.py
@@ -156,7 +156,7 @@ class ReferenceFrame:
         >>> P = ReferenceFrame('P', indices=['1', '2', '3'])
         >>> P.xx
         (P['1']|P['1'])
-        >>> F.zy
+        >>> P.zy
         (P['3']|P['2'])
 
         Unit dyadic is also accessible via the ``u`` attribute:

--- a/sympy/physics/vector/frame.py
+++ b/sympy/physics/vector/frame.py
@@ -145,7 +145,7 @@ class ReferenceFrame:
         >>> type(A) == type(D)
         True
 
-        Unit dyads for the ReferenceFrame can be accessed through the attributes xx, xy, etc. For example:
+        Unit dyads for the ReferenceFrame can be accessed through the attributes ``xx``, ``xy``, etc. For example:
 
         >>> from sympy.physics.vector import ReferenceFrame
         >>> N = ReferenceFrame('N')
@@ -153,11 +153,22 @@ class ReferenceFrame:
         (N.y|N.z)
         >>> N.zx
         (N.z|N.x)
-        >>> F = ReferenceFrame('F', indices=['1', '2', '3'])
-        >>> F.xx
-        (F['1']|F['1'])
+        >>> P = ReferenceFrame('P', indices=['1', '2', '3'])
+        >>> P.xx
+        (P['1']|P['1'])
         >>> F.zy
-        (F['3']|F['2'])
+        (P['3']|P['2'])
+
+        Unit dyadic is also accessible via the ``u`` attribute:
+
+        >>> from sympy.physics.vector import ReferenceFrame
+        >>> N = ReferenceFrame('N')
+        >>> N.unit
+        (N.x|N.x) + (N.y|N.y) + (N.z|N.z)
+        >>> P = ReferenceFrame('P', indices=['1', '2', '3'])
+        >>> P.u
+        (P['1']|P['1']) + (P['2']|P['2']) + (P['3']|P['3'])
+
         """
 
         if not isinstance(name, str):
@@ -1445,6 +1456,11 @@ class ReferenceFrame:
     def zz(self):
         """Unit dyad of basis Vectors z and z for the ReferenceFrame."""
         return Vector.outer(self.z, self.z)
+
+    @property
+    def u(self):
+        """Unit dyadic for the ReferenceFrame."""
+        return self.xx + self.yy + self.zz
 
     def partial_velocity(self, frame, *gen_speeds):
         """Returns the partial angular velocities of this frame in the given

--- a/sympy/physics/vector/tests/test_frame.py
+++ b/sympy/physics/vector/tests/test_frame.py
@@ -658,3 +658,63 @@ def test_dcm_cache_dict():
     assert A._dcm_dict == A._dcm_cache
     assert B._dcm_dict == {C: Matrix([[1, 0, 0],[0, cos(b), -sin(b)],[0, sin(b),  cos(b)]]), \
         A: Matrix([[1, 0, 0],[0, cos(b), -sin(b)],[0, sin(b),  cos(b)]])}
+
+def test_xx_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.xx == Vector.outer(N.x, N.x)
+    assert F.xx == Vector.outer(F.x, F.x)
+
+def test_xy_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.xy == Vector.outer(N.x, N.y)
+    assert F.xy == Vector.outer(F.x, F.y)
+
+def test_xz_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.xz == Vector.outer(N.x, N.z)
+    assert F.xz == Vector.outer(F.x, F.z)
+
+def test_yx_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.yx == Vector.outer(N.y, N.x)
+    assert F.yx == Vector.outer(F.y, F.x)
+
+def test_yy_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.yy == Vector.outer(N.y, N.y)
+    assert F.yy == Vector.outer(F.y, F.y)
+
+def test_yz_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.yz == Vector.outer(N.y, N.z)
+    assert F.yz == Vector.outer(F.y, F.z)
+
+def test_zx_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.zx == Vector.outer(N.z, N.x)
+    assert F.zx == Vector.outer(F.z, F.x)
+
+def test_zy_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.zy == Vector.outer(N.z, N.y)
+    assert F.zy == Vector.outer(F.z, F.y)
+
+def test_zz_dyad():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.zz == Vector.outer(N.z, N.z)
+    assert F.zz == Vector.outer(F.z, F.z)
+
+def test_unit_dyadic():
+    N = ReferenceFrame('N')
+    F = ReferenceFrame('F', indices=['1', '2', '3'])
+    assert N.u == N.xx + N.yy + N.zz
+    assert F.u == F.xx + F.yy + F.zz

